src/Moq/Guard.cs

```diff
@@ -16,6 +16,18 @@ namespace Moq
 	[DebuggerStepThrough]
 	internal static class Guard
 	{
+		public static void HasDefaultConstructor(Type type)
```

Comment:
This might be useful in `MockDefaultValueProvider`, too. Today, `DefaultValue.Mock` will simply attempt to mock types even if they don't have a default constructor, resulting in a DynamicProxy exception (which could be hidden).

An alternative would be to instead add a `type.HasDefaultConstructor()` check inside `type.IsMockable()`. That way, `MockDefaultValueProvider` would no longer attempt to mock default ctor-less types and delegate to `DefaultValue.Empty`.